vm-validator/src/vm_validator.rs

```diff
@@ -72,6 +77,19 @@ impl TransactionValidation for VMValidator {
         .await
         .map_err(Into::into)
     }
+
+    fn restart(&mut self, config: OnChainConfigPayload) -> Result<()> {
+        let gas_schedule = self.vm.get_gas_schedule()?;
+        let publishing_options = config.get::<VMPublishingOption>()?;
+        let version = self.vm.get_libra_version()?;
```

Comment:
Should we obtain this version from `config` as well?

Reply:
is the migration complete for this config?